presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -238,7 +238,7 @@ valueExpression
 primaryExpression
     : NULL                                                                           #nullLiteral
     | interval                                                                       #intervalLiteral
-    | identifier STRING                                                              #typeConstructor
+    | type STRING                                                                    #typeConstructor
```

Comment:
I'm not sure we want to allow arbitrary types for the "type constructor" idiom. I actually think it was a mistake to allow anything beyond the special cases described in the SQL spec (INTERVAL, etc).